### DISCUSSION TITLE
PDE - Maven integration - Should target resolution fail if repo not found?

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-m2e/m2e-core#488] [Q&A] [answered]
A client of mine starts to use the Maven repository integration provided by #202. Current behavior is that if the artifact is not found in the specified repo the system uses Maven central as source. 

Is that intended? Or should the target resolution fail?

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<?pde version="3.8"?>
<target name="target-platform">
	<locations>
		<location includeDependencyDepth="none" includeDependencyScope="compile" includeSource="true" missingManifest="generate" type="Maven">

			<feature id="test" label="Libraries" version="1.0.0.qualifier">

</feature>
			<dependencies>
				<!-- https://mvnrepository.com/artifact/com.beust/jcommander -->
				<dependency>
    				<groupId>com.beust</groupId>
    				<artifactId>jcommander</artifactId>
    				<version>1.3</version>
				</dependency>
			</dependencies>
			<repositories>
			 <repository>
   				<id>central</id>
   				<name>Internal Mirror of Central Repository</name>
   				<url>http://localhost:8081/NOPE/</url>
  				</repository>
			</repositories>
		</location>
	</locations>
</target>
```
I think both behavior would be fine, if the entry is just an additional source, we could restrict the available repository in our master pom but I assumed that the specified repo would be a hard restriction.

cc @HannesWell

**Comment by laeubi** [comment, marked as answer]
Repositories specified in the target are used in addition to the "global" ones. If you want to prevent contacting Maven central (e.g. using a mirror proxy) you should configure this in the global maven settings.

**Reply by HannesWell** [reply]
I think this is a good default and how Maven behaves itself. I'm not even sure if explicitly specified repos in a pom are queried first or the global ones. Furthermore I don't know a use-case where one want's a different behavior unless for example the artifacts are modified in the specified location?
However if there is a strong demand I wonder if it would be possible to add an option to only consider the specified repo. IIRC the M2E Maven Target uses the usual Maven Resolver? Do you know if that resolver can be set up to only consider a specific repo?

But on the other hand we should not add too many options for each possible cause for "just in case".

**Reply by laeubi** [reply]
We should not add options without a strong use-case and I think the description is clear here:

![grafik](https://user-images.githubusercontent.com/1331477/148934640-413ccdf8-6746-4627-82d8-b0eaac3474c6.png)

if one likes to have mirrors it is best to be defined at the maven level with a settings.xml

@vogella if your client likes the maven integration they might consider say [thank you](https://github.com/sponsors/laeubi?frequency=one-time) or even [sponsor further development ](https://github.com/sponsors/laeubi?frequency=recurring) of that feature.
